transport-classes-io_uring/src/main/java/io/netty/channel/uring/AbstractIoUringStreamChannel.java

```diff
@@ -283,7 +319,62 @@ protected int scheduleWriteSingle(Object msg) {
             if (writeId == 0) {
                 return 0;
             }
-            return 1;
+            return outstandingWrite;
+        }
+
+        IoUringIoOps trySendZC(ByteBuf buf) {
+
+            if (!IoUring.isIOUringSendZCSupported()) {
+                return null;
+            }
+
+            IoUringStreamChannelConfig ioUringStreamChannelConfig = (IoUringStreamChannelConfig) config();
+            int fd = fd().intValue();
+
+            if (buf.nioBufferCount() == 1) {
+                int waitSend = buf.readableBytes();
+                if (waitSend != 0
+                    && IoUringStreamChannelConfig.needIOUringSendZC(ioUringStreamChannelConfig, waitSend)) {
+                    long memoryAddress ;
+                   if (buf.hasMemoryAddress()) {
+                        memoryAddress = buf.memoryAddress() + buf.readerIndex();
+                    } else {
+                        ByteBuffer nioBuffer = buf.internalNioBuffer(buf.readerIndex(), buf.readableBytes());
+                        memoryAddress = Buffer.memoryAddress(nioBuffer) + nioBuffer.position();
+                    }
+                    return IoUringIoOps.newSendZC(
+                            fd, memoryAddress, buf.readableBytes(),
+                            0,
+                            nextOpsId(),
+                            0
+                    );
+                } else {
+                    // Not reaching the threshold, so we dont send ZC
+                    return null;
+                }
+            }
+
+            // get the first ByteBuffer
+            // because UnixChannelUtil::isBufferCopyNeededForWrite in AbstractIoUringChannel::filterOutboundMessage
+            // so its hard to reach here
+            ByteBuffer[] byteBuffers = buf.nioBuffers();
```

Comment:
@normanmaurer here,I implemented the code here with reference to the IovArray, so I'm not quite sure if such a scenario will actually occur here. I used CompositeByteBuf for testing and found that, in fact, the branch here cannot be covered at all by AbstractIoUringChannel::filterOutboundMessage.